Jenkinsfile

```diff
@@ -320,6 +327,24 @@ stage('Default build') {
 					${enableDefaultEnvIT ? '' : '-DskipITs'} \\
 					${enableDefaultEnvLegacyIT ? '-Dsurefire.legacy.skip=false -Dfailsafe.legacy.skip=false' : ''}
 			"""
+
+			try {
+				withCredentials([string(credentialsId: 'coveralls-repository-token', variable: 'COVERALLS_TOKEN')]) {
+					sh """ \\
+							mvn coveralls:report \\
+							-DrepoToken=${COVERALLS_TOKEN} \\
+							${env.CHANGE_ID ? """ \\
+									-DpullRequest=${env.CHANGE_ID} \\
+							""" : """ \\
+									-Dbranch=${env.BRANCH_NAME} \\
+							"""} \\
+					"""
+				}
+			}
+			catch (CredentialNotFoundException e) {
+				echo "No Coveralls token configured - skipping Coveralls report. Error was: ${e}"
+			}
```

Comment:
You prefer skipping it (and probably it going unnoticed) rather than throwing the exception?

Reply:
It's for the benefit of anyone configuring a job for his fork and not wanting to also set up coveralls, sonar, and so on. For example if you (@gsmet) set up a job for your own fork, I suppose you'll want the job to work out of the box, and only later, maybe, set up additional integrations.

Note that we only ignore the `CredentialNotFoundException`. As long as you provide credentials, errors should be reported correctly and fail the build. Which means in particular that, for the job currently configured on master, this `catch` block has no effect.